source/val/validate_capability.cpp

```diff
@@ -67,6 +67,47 @@ bool IsSupportGuaranteedVulkan_1_2(uint32_t capability) {
   return false;
 }
 
+bool IsSupportGuaranteedVulkan_1_3(uint32_t capability) {
+  if (IsSupportGuaranteedVulkan_1_2(capability)) return true;
+  switch (spv::Capability(capability)) {
+    case spv::Capability::DotProduct:
+    case spv::Capability::DotProductInputAll:
+    case spv::Capability::DotProductInput4x8Bit:
+    case spv::Capability::DotProductInput4x8BitPacked:
+    case spv::Capability::VulkanMemoryModel:
+    case spv::Capability::VulkanMemoryModelDeviceScope:
+    case spv::Capability::PhysicalStorageBufferAddresses:
+    case spv::Capability::DemoteToHelperInvocation:
+      return true;
+    default:
+      break;
+  }
+  return false;
+}
+
+bool IsSupportGuaranteedVulkan_1_4(uint32_t capability) {
+  if (IsSupportGuaranteedVulkan_1_3(capability)) return true;
+  switch (spv::Capability(capability)) {
+    case spv::Capability::UniformBufferArrayDynamicIndexing:
+    case spv::Capability::SampledImageArrayDynamicIndexing:
+    case spv::Capability::StorageBufferArrayDynamicIndexing:
+    case spv::Capability::StorageImageArrayDynamicIndexing:
+    case spv::Capability::Int16:
+    case spv::Capability::StorageBuffer16BitAccess:
+    case spv::Capability::VariablePointers:
+    case spv::Capability::VariablePointersStorageBuffer:
+    case spv::Capability::UniformTexelBufferArrayDynamicIndexing:
+    case spv::Capability::StorageTexelBufferArrayDynamicIndexing:
+    case spv::Capability::Int8:
+    case spv::Capability::StorageBuffer8BitAccess:
+    case spv::Capability::FloatControls2:
```

Comment:
From https://docs.vulkan.org/spec/latest/chapters/features.html#:~:text=If%20Vulkan%201.4%20is%20supported%2C%20the%20following%20features%20must%20be%20supported I also get:

*  ExpectAssumeKHR
*  SampleRateShading
* StorageImageExtendedFormats
* ImageGatherExtentded 
* StorageBuffer16BitAccess

Reply:
1.4 didn't have a new SPIR-V version so I skipped any capabilities that require an extension anyways (so ExpectAssumeKHR).

16-bit storage is further up the list. The other three I'll add.